packages/data-stores/src/plans/test/resolvers.ts

```diff
@@ -14,69 +34,10 @@ describe( 'getSupportedPlans', () => {
 	it( 'calls setFeatures, setFeaturesByType, and setPlans after fetching plans', () => {
 		const iter = getSupportedPlans();
 
-		const planPriceData = [
-			{
-				currency_code: 'INR',
-				product_slug: PLAN_FREE,
-				raw_price: 0,
-				product_id: 1,
-			},
-			{
-				// premium plan, billed annually
-				currency_code: 'INR',
-				product_slug: PLAN_PREMIUM,
-				path_slug: 'premium',
-				raw_price: 12,
-				product_id: 2,
-			},
-			{
-				// premium plan, billed monthly
-				currency_code: 'INR',
-				product_slug: PLAN_PREMIUM_MONTHLY,
-				raw_price: 13,
-				product_id: 3,
-				bill_period: 31,
-			},
-		];
-
-		const planDetailedData = {
-			groups: [],
-			plans: [
-				{
-					products: [
-						{
-							plan_id: 1,
-						},
-						{
-							plan_id: 2,
-						},
-						{
-							plan_id: 3,
-						},
-					],
-					features: [ 'custom-domain' ],
-					highlighted_features: [ 'Custom domain', 'Other feature' ],
-					nonlocalized_short_name: 'Premium',
-				},
-			],
-			features_by_type: [
-				{
-					id: 'general',
-					name: null,
-					features: [ 'custom-domain', 'other-feature' ],
-				},
-			],
-			features: [
-				{
-					id: 'custom-domain',
-					name: 'Custom domain',
-				},
-				{
-					id: 'other-feature',
-					name: 'Other feature',
-				},
-			],
-		};
+		// Prepare stricter iterator types
+		type IteratorReturnType = ReturnType< typeof iter.next >;
+		type PlanPriceApiDataIterator = ( planPriceData: PricedAPIPlan[] ) => IteratorReturnType;
+		type PlanDetailsApiDataIterator = ( { body: DetailsAPIResponse } ) => IteratorReturnType;
```

Comment:
This is to "help" TypeScript to expect the correct arguments when invoking `iter.next()` later on in the tests